webdriver/tests/element_click/element_click.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+
+from tests.support.asserts import assert_error, assert_success
+from tests.support.inline import inline
+
+def click(session, element):
+    return session.transport.send("POST", "session/{session_id}/element/{element_id}/click"
+                                  .format(session_id=session.session_id,
+                                          element_id=element.id))
+
+
+# 14.1 Element Click
+
+def test_no_browsing_context(session, create_window):
+    # 14.1 step 1
+    session.window_handle = create_window()
+    session.close()
+
+    response = session.transport.send("POST", "session/{session_id}/element/{element_id}/click"
+                                      .format(session_id=session.session_id,
+                                              element_id="foo"))
+    assert_error(response, "no such window")
+
+
+def test_scroll_into_element_view(session):
+    # 14.1 Step 4
+    session.url = inline("<input type=text value=Federer><div style=\"height: 200vh; width: 5000vh\">")
+
+    # Scroll to the bottom right of the page
+    session.execute_script("window.scrollTo(document.body.scrollWidth, document.body.scrollHeight);")
+    element = session.find.css("input", all=False)
+    # Clear and scroll back to the top of the page
+    response = click(session, element)
+    assert_success(response)
+
+    # Check if element cleared is scrolled into view
+    response = session.execute_script("""rect = document.getElementsByTagName("input")[0].getBoundingClientRect();
+                                        xOffset = window.pageXOffset;
+                                        yOffset = window.pageYOffset;
+                                        return  rect["top"] < (window.innerHeight + yOffset) &&
+                                                rect["left"] < (window.innerWidth + xOffset) &&
+                                                (rect["top"] + rect["height"]) > yOffset &&
+                                                (rect["left"] + rect["width"]) > xOffset;""")
+    assert response is True
+
+
+@pytest.mark.parametrize("transform", ["translate(-50px, -50px)", "rotate(50deg)"])
+
+def test_element_not_interactable(session, transform):
+    # 14.1 Step 5
+    session.url = inline("<div style=\"width: 500px; height: 100px; background-color: green; transform: " + transform + """;">
+                            <a href=#>aaaa</a>
```

Comment:
I don’t understand what the link is for.

Reply:
I am trying to click that link. The div is to show that the element is not interactable because it is out of view.

Reply:
Right, so why not make the `<div>` have a click handler? This seems to conflate testing navigation to fragment, which is undesirable.

Reply:
fixed